[1.16] Add docs on DAPR_UNSAFE_SKIP_CONTAINER_UID_GID_CHECK env var

JoshVanL opened the pull request:
Fixes https://github.com/dapr/docs/issues/4784

JoshVanL committed:
[1.16] Add docs on DAPR_UNSAFE_SKIP_CONTAINER_UID_GID_CHECK env var

Fixes https://github.com/dapr/docs/issues/4784

Signed-off-by: joshvanl <me@joshvanl.dev>

msfussell commented on daprdocs/content/en/concepts/security-concept.md:
```diff
@@ -198,6 +198,15 @@ Let's go over each of the security capabilities and describe how they are protec
 8. The data in the Redis state store is encrypted at rest and can only be read using the correctly configured Dapr Redis state store component.
 
+## Run as non-root
+When running in Kubernetes, Dapr services will ensure each process is running as non-root.
```
```suggestion
When running in Kubernetes, Dapr services ensure each process is running as non-root.
```

msfussell commented on daprdocs/content/en/operations/hosting/kubernetes/kubernetes-production.md:
```diff
@@ -343,10 +343,20 @@ By default, the Dapr sidecar injector injects a sidecar without any `seccompProf
 
-## Best Practices 
+## Run as non-root
+When running in Kubernetes, Dapr services will ensure each process is running as non-root.
```
```suggestion
When running in Kubernetes, Dapr services ensure each process is running as non-root.
```

msfussell reviewed (commented):
@JoshVanL - Could you also add this to the Env Variables reference doc here https://docs.dapr.io/reference/environment/

JoshVanL committed:
Apply suggestions from code review

Co-authored-by: Mark Fussell <markfussell@gmail.com>
Signed-off-by: Josh van Leeuwen <me@joshvanl.dev>

JoshVanL committed:
Merge branch 'v1.16' into DAPR_UNSAFE_SKIP_CONTAINER_UID_GID_CHECK

JoshVanL committed:
Adds DAPR_UNSAFE_SKIP_CONTAINER_UID_GID_CHECK to environment reference

Signed-off-by: joshvanl <me@joshvanl.dev>

JoshVanL commented:
@msfussell done!

msfussell committed:
Moved non-root execution section for Kubernetes

Moved the section on running Dapr services as non-root in Kubernetes, including environment variable configuration.

Signed-off-by: Mark Fussell <markfussell@gmail.com>

msfussell reviewed (approved):
LGTM

msfussell committed:
Merge branch 'v1.16' into DAPR_UNSAFE_SKIP_CONTAINER_UID_GID_CHECK
